clients/stellar-relay-lib/src/connection/connector/message_reader.rs

```diff
@@ -35,37 +31,56 @@ pub(crate) async fn poll_messages_from_stellar(
 		// check for messages from user.
 		match send_to_node_receiver.try_recv() {
 			Ok(msg) =>
-				if let Err(e) = connector.send_to_node(msg).await {
+				if let Err(e) = connector.send_to_node(msg) {
 					log::error!("poll_messages_from_stellar(): Error occurred during sending message to node: {e:?}");
 				},
-			Err(TryRecvError::Disconnected) => break,
+			Err(TryRecvError::Disconnected) => {
+				log::trace!("poll_messages_from_stellar(): Recv channel (for sending message to node) got disconnected.");
+				break
+			},
 			Err(TryRecvError::Empty) => {},
 		}
 
+		// clone the stream to perform a read operation on the next function calls
+		let stream_clone =
+			match connector.tcp_stream.try_clone() {
+				Err(e) => {
+					log::error!("poll_messages_from_stellar(): Error occurred during cloning tcp stream: {e:?}");
+					break
+				},
+				Ok(stream_clone) => stream_clone,
+			};
+
 		// check for messages from Stellar Node.
-		match read_message_from_stellar(&mut read_stream_overlay, connector.timeout_in_secs).await {
+		let xdr = match read_message_from_stellar(stream_clone) {
```

Comment:
Would it make sense here to use `tokio::task::spawn_blocking` or `std::thread::spawn` ? I imagine this reading operation could take a while if the network has delays, or the stellar node is slow, etc. and it may block the thread. With the `async` version of the `TcpStream` this wasn't a concern.

For message passing `mpsc::unbounded_channel` apparently would [work](https://stackoverflow.com/questions/61168507/wrapping-blocking-mpsc-in-async-rust-tokio).

Or perhaps this is overkill for this application and we are fine with blocking?